### ISSUE TITLE
Overlapping equality checking

### ISSUE THREAD

**Issue by jgarvin** [microsoft/pyright#5919]
mypy helpfully rejects:

```
if 3 == "hello":
    print("hi")
```

But pyright does not. Ideally this would also work for match/case.

**Comment by erictraut** [comment]
Pyright does detect both cases if you enable `reportUnnecessaryComparison`. This check is enabled in strict mode.

```python
# pyright: reportUnnecessaryComparison=true

match 3:
    case "hello":
        pass

if 3 == "hello":
    pass
```

![image](https://github.com/microsoft/pyright/assets/7040122/a8e0f2e5-53bb-4a27-b460-6f54fcdab290)

**Comment by jgarvin** [comment]
ah, I tried using `--strict` like mypy but apparently that's not how you turn it on, thanks
